src/api/firebase.js

```diff
@@ -70,7 +74,28 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 	});
 }
 
-export async function updateItem() {
+export async function updateItem(listId, itemId, checked) {
+	const listItemRef = doc(db, listId, itemId);
+	const listItemSnap = await getDoc(listItemRef);
+	const currentTotalPurchases = listItemSnap.data().totalPurchases;
+	const currentTime = listItemSnap.data().dateLastPurchased;
+
+	let totalPurchases = currentTotalPurchases;
+	let dateLastPurchased = currentTime;
```

Comment:
I'm not quite sure I'm following the naming of your variables here. Why is `currentTime` being set to `dateLastPurchased` above, then `dateLastPurchased` set to `currentTime`?